src/ILCompiler.Compiler/src/Compiler/CombinedCompilationRootProvider.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace ILCompiler
+{
+    /// <summary>
+    /// Set of compilation roots that is a result of multiple <see cref="CompilationRootProvider"/> instances.
+    /// </summary>
+    public class CombinedCompilationRootProvider : CompilationRootProvider
```

Comment:
Would it look better if the compilation just takes `IEnumerable<CompilationRootProvider>` instead of this wrapper?